packages/react-cookie/src/tests/CookieUniversalProvider.test.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import {createMount} from '@shopify/react-testing';
+import {
+  NetworkManager,
+  NetworkContext,
+  ServerCookieManager,
+} from '@shopify/react-network';
+import {CookieUniversalProvider} from '../CookieUniversalProvider';
+import {CookieContext} from '../context';
+import {BrowserCookieManager} from '../BrowserCookieManager';
+
+const {hasDocumentCookie} = require.requireMock('../utilities');
+
+jest.mock('../utilities', () => ({
+  hasDocumentCookie: jest.fn(),
+}));
+
+describe('CookieUniversalProvider', () => {
+  beforeEach(() => {
+    hasDocumentCookie.mockImplementation(() => true);
+  });
+
+  afterEach(() => {
+    hasDocumentCookie.mockClear();
+  });
+
+  function MockApp() {
+    return null;
+  }
+
+  it('provides a universal cookie manager for the server', () => {
+    hasDocumentCookie.mockImplementation(false);
```

Comment:
we might want to add a 'why' comment explaining why we have to mock this (so that we can test the server implementation despite jest always having a document.cookie defined)